tests/std/tests/LWG3561_discard_block_engine_counter/test.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <iosfwd>
+#include <random>
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
+using namespace std;
+
+constexpr size_t large_block_size = static_cast<size_t>(UINT_MAX) - 16;
+constexpr size_t large_used_block = static_cast<size_t>(UINT_MAX) - 20;

```

Comment:
These `static_cast`s are unnecessary. We're already saying that the `constexpr` variables are `size_t`, and the initializers are constant expressions.